cmake/scripts/common/AddonHelpers.cmake

```diff
@@ -41,6 +41,20 @@ endmacro()
 macro (build_addon target prefix libs)
   addon_version(${target} ${prefix})
 
+  # Check addons version is equal or less to Kodi's Version
+  # 1. If addon version higher as Kodi version, seems it not to correct API and
+  #    need to abort, e.g. to prevent on new addon release an upload to wrong
+  #    Kodi addon mirror.
+  # 2. If addon version is lower can it maybe still work and is further usable,
+  #    as there maybe no API changes since them.
+  #    Bring only a warning message in this case.
+  string(REGEX REPLACE "^([0-9]*)\.[0-9]*\.[0-9]*" "\\1" ${prefix}_VERSION_MAJOR ${${prefix}_VERSION})
+  if(${${prefix}_VERSION_MAJOR} GREATER ${APP_VERSION_MAJOR})
+    message(FATAL_ERROR "Addon major version ${${prefix}_VERSION_MAJOR} higher as Kodi's version ${APP_VERSION_MAJOR}. In this case, correctly used API cannot be guaranteed and construction will be aborted!")
+  elseif(${${prefix}_VERSION_MAJOR} LESS ${APP_VERSION_MAJOR})
+    message(WARNING "Addon major version ${${prefix}_VERSION_MAJOR} smaller as Kodi's version ${APP_VERSION_MAJOR}. Attention: Possible API changes may not be included. Since Kodi 20, addon and Kodi should use the same major version.")
```

Comment:
```
    message(WARNING "Addon major version ${${prefix}_VERSION_MAJOR} smaller than Kodi's major version ${APP_VERSION_MAJOR}. WARNING: Possible API changes may not be included. Since Kodi 20, addon and Kodi should use the same major version.")
```

Reply:
Have updated, thanks for help about!